contracts/operators/StakeDAO/StakeDaoCurveStrategyOperator.sol

```diff
@@ -0,0 +1,379 @@
+import "../../interfaces/external/ICurvePool/ICurvePoolNonETH.sol";
+
+/// @title StakeDAO Curve strategy operator
+/// @notice Deposit/Withdraw in a StakeDAO strategy
+contract StakeDaoCurveStrategyOperator {
+    StakeDaoStrategyStorage public immutable operatorStorage;
+
+    /// @dev ETH address
+    address public immutable eth;
+
+    /// @dev WETH contract
+    IWETH private immutable weth;
+
+    /// @dev Withdrawer
+    Withdrawer private immutable withdrawer;
+
+    constructor(
+        address[] memory strategies,
+        CurvePool[] memory pools,
+        Withdrawer _withdrawer,
+        address _eth,
+        address _weth
+    ) {
+        uint256 strategiesLength = strategies.length;
+        require(strategiesLength == pools.length, "SDCSO: INVALID_POOLS_LENGTH");
+        operatorStorage = new StakeDaoStrategyStorage();
+
+        for (uint256 i; i < strategiesLength; i++) {
+            operatorStorage.addStrategy(strategies[i], pools[i]);
+        }
+
+        operatorStorage.transferOwnership(msg.sender);
+
+        eth = _eth;
+        weth = IWETH(_weth);
+        withdrawer = _withdrawer;
+    }
+
+    /// @notice Add liquidity in a Curve pool that inculde ETH,
+    ///         deposit the LP token in a StaeDAO strategy and receive
+    ///         the StaeDAO strategy shares
+    /// @param strategy The StaeDAO strategy address to deposit into
+    /// @param amount The amount of token to add liquidity
+    /// @param minStrategyAmount The minimum of StaeDAO strategy shares expected
+    /// @return amounts Array of amounts :
+    ///         - [0] : The strategy token received amount
+    ///         - [1] : The token deposited amount
+    /// @return tokens Array of token addresses
+    ///         - [0] : The strategy token received address
+    ///         - [1] : The token deposited address
+    function depositETH(
+        address strategy,
+        uint256 amount,
+        uint256 minStrategyAmount
+    ) public payable returns (uint256[] memory amounts, address[] memory tokens) {
+        require(amount != 0, "YCVO: INVALID_AMOUNT");
+
+        (address pool, uint96 poolCoinAmount, address lpToken) = operatorStorage.strategies(strategy);
+        require(pool != address(0), "YCVO: INVALID_STRATEGY");
+
+        uint256 strategyBalanceBefore = IERC20(strategy).balanceOf(address(this));
+        uint256 ethBalanceBefore = weth.balanceOf(address(this));
+
+        _addLiquidityAndDepositETH(strategy, ICurvePoolETH(pool), IERC20(lpToken), poolCoinAmount, amount);
+
+        (amounts, tokens) = CurveHelpers.getOutputAmounts(
+            IERC20(address(weth)),
+            ethBalanceBefore,
+            amount,
+            IERC20(strategy),
+            strategyBalanceBefore,
+            minStrategyAmount
+        );
+    }
+
+    /// @notice Add liquidity to a Curve pool using the input token,
+    ///         deposit the LP token in a StakeDAO strategy and receive
+    ///         the strategy's token.
```

Comment:
In `depositETH` we talk about shares and here we talk about token 
We should standardize the wording